### PR TITLE
Add description field to proposals and agendaitems

### LINKED ISSUE #4229: Beschreibung für Traktanden/Anträge
Gewünscht ist die Möglichkeit eine Beschreibung Anträge/Traktanden hinzuzufügen. 
- Erhöht die Übersichtlichkeit der Traktandenliste (Vermeidung langer Titel)
- Beschreibung als Metadatum in Plone, analog zu anderen Inhaltstypen 

Ergänzung Antrag/Eingereichter Antrag:
- Feld Beschreibung hinzufügen
- Add/Edit Formular aktualisieren
- Beschreibung in Übersichts-Tab und Listings aufnehmen

screen shot 2018-04-25 at 11 04 38

Ergänzungen Traktandum (Ad-Hoc):
- Beschreibung als Spalte definieren
- Beschreibung für Ad-Hoc Traktanden beim Traktandieren

traktandieren mit beschreibung - page 1

Sonstinge Ergänzungen Sitzungsansicht:
- Beschreibung beim Traktandieren von Anträgen darstellen
- Aktion umbenennen: `Traktandum umbenennen` -> `Traktandum bearbeiten`
- Maske Traktandum bearbeiten ergänzen, Beschreibungsfeld anbieten

screen shot 2018-04-25 at 11 17 37

Ergänzung Sablon:
- Beschreibung als Sablon-Variable von Anträgen zur Verfügung stellen (Traktandenliste, Protokoll-Kopf, ...)

Ref: https://basecamp.com/2768704/projects/13732725/todos/340577556#comment_616241599
Rapportieren: `Projekt - Umsetzung GEVER-RIS`

### PR DESCRIPTION
**Pull request by njohner**
* new description field in the proposal schema
* `description` and `submitted_description` columns in the proposal model (these are needed for the listings, as the listings are made from a query to the SQL database)
* `description` column in the agendaitem model (again, needed for the listings)
* Adapted fixture and tests to include proposal descriptions
* Include description when adding an ad-hoc `AgendaItem`
* Include description when editing an `AgendaItem`
* Rename action from `Traktandum umbenennen` to `Traktandum bearbeiten`
* Update `examplecontent` sablon templates to include descriptions for proposals and agendaitems
* Add proposal description to docproperties

❗️ `description` property was removed from the `model.agendaitem`. `description` is now the field, and a `get_description` method was added, which does only return the `description` field if the `agendaitem` has no `proposal` (i.e. is an ad-hoc agendaitem).

**description field in proposal add form**
screen shot 2018-06-19 at 12 36 11

**In the proposal overview**
screen shot 2018-06-19 at 12 37 16

**In the Proposal tab**
screen shot 2018-06-19 at 12 38 20

**In the meeting view, description is shown for submitted proposals that can be scheduled and for scheduled agendaitems. There is also a description field when adding an ad-hoc AgendaItem**
screen shot 2018-06-19 at 13 33 41

**Form to modify an agendaitem**
screen shot 2018-06-19 at 13 23 30

resolves #4229

### CODE REVIEW

**Comment by njohner** [comment]
Ok all done!
For the small comments you had in the code, I integrated them directly in the respective commits.
For the other points you raised in your review, I made specific commits to address each one (starting with `Move description field in edit form`). If/when you are satisfied with the work, I'll squash them into the appropriate commits.
There are two commits that are for errors I discovered while making these changes:
* `Prefill edit form with values from SubmittedProposal`: The edit form Title and Description fields were prefilled with the `title` and `description` of the model, hence of the `Proposal`, instead of the `SubmittedProposal`
* `Display submitted_description in proposal listing in committee`. The proposal listing in the committee displayed the `description` instead of the `submitted_description` column.

For your comment on the description wrapping around the buttons, instead of limiting the width of the description, I clear the space above it. I think that looks better when the window is not that wide, (see screenshots below). The behaviour of the title is to wrap around the buttons though, and I dont see a very good solution there.

screen shot 2018-06-26 at 14 45 48
screen shot 2018-06-26 at 14 46 14

**Comment by njohner** [comment]
Right, html and linebreak was taken care of only for the proposal overview, not the agendaitem list. This is now fixed.

File: opengever/meeting/browser/resources/meeting.js [hunk 1]
```diff
@@ -434,7 +434,9 @@
       };
 
       self.submit = function() {
-        var data = {title: self.title_field.val().trim(), description: self.description_field.val().trim()};
+        var escaped_title = $('<div>').text(self.title_field.val().trim()).html();
```
Comment (deiferni) [inline opengever/meeting/browser/resources/meeting.js]: this must happen server-side, on render. never trust the client! this way a malicious user can submit on his own an thus bypass your escaping.

for an example you can have a look at dossier comments and how it is implemented there
https://github.com/4teamwork/opengever.core/blob/cd1d52a43c84059941e1b2863b140207a469080d/opengever/dossier/base.py#L378-L386

Comment (njohner) [reply opengever/meeting/browser/resources/meeting.js]: Right that makes sense...

**Comment by njohner** [comment]
Ok I think this is ready.
For the unscheduled proposals list, the styling is here: https://github.com/4teamwork/opengever.core/pull/4471/files#diff-56a570d675a02ccf9601a25fc546b1b6R1121
and the relevant method call is here:
https://github.com/4teamwork/opengever.core/blob/99981664893ae9022a8ac14fc88565dbfb1a38da/opengever/meeting/browser/meetings/unscheduled_proposals.py#L62

**Comment by njohner** [comment]
Ready for re-re-re-re-re-review.
I found one more error in handling of html tags in agendaitem titles. 
I also adapted the UI to the changes introduced in https://github.com/4teamwork/opengever.core/pull/4490, specifically moved the `submit` button below the table used to choose the template

screen shot 2018-07-02 at 16 37 29

File: plonetheme/teamraum/resources/css/gever/partials/_meeting.scss [hunk 2]
```diff
@@ -1110,6 +1113,21 @@ table#agenda_items {
     margin-top: 2em;
     margin-left: 0em;
   }
+
+}
+
+#unscheduled_porposals.list-group {
+  .submit{
+    display:flex;
```
Comment (deiferni) [inline plonetheme/teamraum/resources/css/gever/partials/_meeting.scss]: missing whitespace 🙄

**Comment by njohner** [comment]
Ok I think the form is looking quite a bit better now...

screen shot 2018-07-03 at 15 02 52

screen shot 2018-07-03 at 15 03 03

**Pull request merged**